lib/discovery-indexer/reader/purlxml_parser_strict.rb

```diff
@@ -142,14 +143,31 @@ def parse_dor_content_type
         dct
       end
 
-      # the @id attribute of resource/file elements that match the image type, including extension
+      # the @id attribute of resource/file elements that match the image, page, or thumb resource type, including extension
       # @return [Array<String>] filenames
       def parse_image_ids
         content_md = parse_content_metadata
-        return nil if content_md.nil?
+        return [] if content_md.nil?
         content_md.xpath('//resource[@type="page" or @type="image" or @type="thumb"]/file[@mimetype="image/jp2"]/@id').map(&:to_s)
       end
 
+      # the druid and id attribute of resource/file and objectId and fileId of the
+      # resource/externalFile elements that match the image, page, or thumb resource type, including extension
+      # Also, prepends the corresponding druid and %2F specifically for Searchworks use
+      # @return [Array<String>] filenames
+      def parse_sw_image_ids
+        content_md = parse_content_metadata
+        return [] if content_md.nil?
+        ids = []
+        content_md.xpath('//resource[@type="page" or @type="image" or @type="thumb"]').each do |node|
+          ids = node.xpath('//file[@mimetype="image/jp2"]/@id').map{ |x| "#{@druid}%2F" + x }
```

Comment:
actually, there are 9 jp2s that should be in the expected results for parse_sw_image_ids and 6 in the parse_image_ids, which is what I put in my test, so the implementation is correct, I think.